src/everwillow/parameters/transforms.py

```diff
@@ -70,41 +73,35 @@ class MinuitTransform(AbstractParameterTransformation):
     upper: ArrayLike
 
     def __post_init__(self):
-        self.lower = eqx.error_if(
-            float_array(self.lower),
-            ~jnp.isfinite(self.lower),
-            "lower bound must be finite.",
-        )
-        self.upper = eqx.error_if(
-            float_array(self.upper),
-            ~jnp.isfinite(self.upper),
-            "upper bound must be finite.",
-        )
-        self.lower, self.upper = eqx.error_if(
-            (self.lower, self.upper),
-            self.lower >= self.upper,
-            f"{self} requires lower bound to be strictly less than upper bound.",
-        )
+        if not jnp.isfinite(self.lower):
+            raise ValueError("lower bound must be finite.")
+        if not jnp.isfinite(self.upper):
+            raise ValueError("upper bound must be finite.")
+        if self.lower >= self.upper:
+            msg = f"{self} requires lower bound to be strictly less than upper bound."
+            raise ValueError(msg)
 
     def unwrap(self, value: ArrayLike) -> ArrayLike:
         """Convert a bounded value into an unconstrained representation."""
-        value = float_array(value)
+        value, lower, upper = map(float_array, (value, self.lower, self.upper))
         value = eqx.error_if(value, ~jnp.isfinite(value), "value must be finite.")
```

Comment:
@pfackeldey  -- follow up from comment above, why the inconsistent treatment here?